pydantic/functional_validators.py

```diff
@@ -423,7 +423,7 @@ def __call__(  # noqa: D102
 ModelAfterValidator = Callable[[_ModelType, _core_schema.ValidationInfo], _ModelType]
 """A `@model_validator` decorated function signature. This is used when `mode='after'`."""
 
-_AnyModelWrapValidator = Union[ModelWrapValidator, ModelWrapValidatorWithoutInfo]
+_AnyModelWrapValidator = Union[ModelWrapValidator[_ModelType], ModelWrapValidatorWithoutInfo[_ModelType]]
```

Comment:
Doesn't this need to be parametrized later?

Reply:
From my testing, adding the `_ModelType` parameter to the overload return type on line 435 seems to give the right results for a correct and incorrect validator (and the `after` overload does the same thing already):
```py
) -> Callable[[_AnyModelWrapValidator[_ModelType]], _decorators.PydanticDescriptorProxy[_decorators.ModelValidatorDecoratorInfo]]:
```
That appears to be the only place where these types are used.